warp10/src/main/java/io/warp10/continuum/gts/INTERPOLATE.java

```diff
@@ -27,23 +25,514 @@
+
+        //
+        // If bucket is on the current tick, advance tick
+        //
+        if (bucket == filled.ticks[i]) {
+          continue;
+        }
+
+        //
+        // Fill missing values until bucket passes the current tick
+        //
+
+        while(bucket < filled.ticks[i]) {
+          if (function.isValidPoint(bucket)) {
+            GTSHelper.setValue(filled, bucket, function.value(bucket));
+          } else {
+            Number filler = (Number) params.get(PARAM_INVALID_TICK_VALUE);
+            if (null != filler) {
+              GTSHelper.setValue(filled, bucket, filler);
+            }
+          }
+          bucket += filled.bucketspan;
+        }
+      }
+
+    } else {
+      // count geo
+      for (int i = 0; i < nvalues; i++) {
+
+        if ((null != filled.elevations) && (GeoTimeSerie.NO_ELEVATION != filled.elevations[i])) {
+          nElevations++;
+        }
+
+        if ((null != filled.locations) && (GeoTimeSerie.NO_LOCATION != filled.locations[i])) {
+          nLocations++;
+        }
+      }
+
+      // fill occurrence ticks
+      for (Long tick: (List<Long>) params.get(PARAM_OCCURRENCES)) {
+        if (function.isValidPoint(tick)) {
+          GTSHelper.setValue(filled, tick, function.value(tick));
+        } else {
+          Number filler = (Number) params.get(PARAM_INVALID_TICK_VALUE);
+          if (null != filler) {
+            GTSHelper.setValue(filled, tick, filler);
+          }
+        }
+      }
+    }
+
+    //
+    // Take care of geo if we have at least two valid values for either elevation or location
+    //
+
+    PolynomialSplineFunction elevFunction = null;
+    PolynomialSplineFunction latFunction = null;
+    PolynomialSplineFunction lonFunction = null;
+
+    //
+    // Compute interpolators
+    //
+
+    double[] xelev = null;
+    double[] felev = null;
+    double[] xloc = null;
+    double[] flat = null;
+    double[] flon = null;
+
+    if (nElevations >= 2) {
+      xelev = new double[nElevations];
+      felev = new double[nElevations];
+    }
+
+    if (nLocations >= 2) {
+      xloc = new double[nLocations];
+      flat = new double[nLocations];
+      flon = new double[nLocations];
+    }
+
+    for (int i = 0; i < nvalues; i++) {
+      if (nElevations >= 2) {
+        if (GeoTimeSerie.NO_ELEVATION != filled.elevations[i]) {
+          xelev[i] = ((Number) GTSHelper.tickAtIndex(filled, i)).doubleValue();
+          felev[i] = ((Number) filled.elevations[i]).doubleValue();
+        }
+      }
+
+      if (nLocations >= 2) {
+        if (GeoTimeSerie.NO_LOCATION != filled.locations[i]) {
+          xloc[i] = ((Number) GTSHelper.tickAtIndex(filled, i)).doubleValue();
+
+          double[] latlon = GeoXPLib.fromGeoXPPoint(filled.locations[i]);
+          flat[i] = latlon[0];
+          flon[i] = latlon[1];
+        }
+      }
+    }
+
+    //
+    // Fill elevation
+    //
+
+    if (nElevations >= 2) {
+      if (null == params.get(PARAM_INTERPOLATOR_GEO)) {
```

Comment:
I don't know if this is usefull, but you can add a new `interpolator.elevation` parameter (if you want linear interpolation for location, and another one for elevation). maybe `interpolator.geo` can become `interpolator.location`